lib/vagrant-openshift/action/build_sti_base_windows.rb

```diff
@@ -0,0 +1,128 @@
+#--
+# Copyright 2016 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#++
+
+require "fog"
+require_relative "../aws"
+
+module Vagrant
+  module Openshift
+    module Action
+      class BuildStiBaseWindows
+        include CommandHelper
+
+        def initialize(app, env, options)
+          @app = app
+          @env = env
+          @options = options
+        end
+
+        def call(env)
+          aws_creds = Vagrant::Openshift::Aws::aws_creds()
+          raise "AWS credentials not configured" if aws_creds.nil?
+          compute = Fog::Compute.new(Vagrant::Openshift::Aws::fog_config(aws_creds))
+
+          @env[:ui].info("Finding base AMI...")
+          images = compute.images.all({"Owner" => "amazon", "name" => "Windows_Server-2012-R2_RTM-English-64Bit-Base-*", "state" => "available"})
+          # https://github.com/fog/fog-aws/issues/320 would make this less hacky
+          images.sort_by! {|image| image.name[-10,10]}
+          @env[:ui].info("Using base AMI #{images.last.name}, #{images.last.id}")
+
+          @env[:ui].info("Creating instance...")
+          instance = compute.servers.create(
+            :tags => {"Name" => "#{@options[:instance_prefix]}-windows2012r2_#{Time.now.to_i}"},
+            :image_id => images.last.id,
+            :flavor_id => "m4.large",
+            :key_name => "libra",
+            :subnet_id => "subnet-cf57c596",
+            :user_data => <<'USERDATA'
```

Comment:
In commands like `build-origin-base` we've moved to having script files in this repo that get deposited on the remote system and executed explicitly. I find that it is a lot less mentally taxing to not have to consider all of the shell script syntax in the context of being an inline Ruby string, etc. If it's a valid approach in this instance, I think it would be a better choice for long-term maintainability of the provision script.